src/sdlsound.cpp

```diff
@@ -190,6 +192,13 @@ void musicFinished()
     Mix_FreeMusic( current_music );
     current_music = nullptr;
 
+    std::string new_playlist = music::get_music_id_string();
+
+    if( current_playlist.compare( new_playlist ) ) {
```

Comment:
```suggestion
    if( current_playlist != new_playlist ) {
```

Reply:
Heh. I'm not familiar with C++ stuff so I did not know that compare method has some issues. I guess I should've just used the comparison operator like a sane person. Applied the changes based on your suggestions.